h2o-algos/src/main/java/hex/tree/SharedTreeModel.java

```diff
@@ -414,4 +413,19 @@ public void generate(JCodeSB out) {
     return (T) sb.p(mname).p("_Forest_").p(t);
   }
 
+  @Override
+  protected void scoreSetupLocal() {
+    Key[][] keys = _output._treeKeys;
+    if (_output._compressedTrees == null || _output._compressedTrees.length < keys.length) {
+      _output._compressedTrees = new CompressedTree[keys.length][];
+      for (int t = 0; t < keys.length; ++t) {
+        _output._compressedTrees[t] = new CompressedTree[keys[t].length];
+        for (int c = 0; c < keys[t].length; ++c) {
+          if (keys[t][c] != null) {
+            _output._compressedTrees[t][c] = DKV.getGet(keys[t][c]);
```

Comment:
we should probably call prefetch on all the keys first, to initiate the remote tasks in parallel